cub/cub/device/device_segmented_sort.cuh

```diff
@@ -1143,17 +1165,30 @@ struct DeviceSegmentedSort
                            EndOffsetIteratorT d_end_offsets,
                            cudaStream_t stream = 0)
   {
-    return SortKeysDescending<KeyT,
-                              BeginOffsetIteratorT,
-                              EndOffsetIteratorT>(d_temp_storage,
-                                                  temp_storage_bytes,
-                                                  d_keys_in,
-                                                  d_keys_out,
-                                                  num_items,
-                                                  num_segments,
-                                                  d_begin_offsets,
-                                                  d_end_offsets,
-                                                  stream);
+    constexpr bool is_descending = true;
+    constexpr bool is_stable = true;
+    constexpr bool is_overwrite_okay = false;
```

Comment:
It seems that you are not using the formatter, could you check whether you have clang-format 17 installed?

Reply:
When I run clang-format I get thousands of lines of diff, which would make the PR impossible to review.
I'm using the dev containers and have clang-format 17.0.6 in the container, but it seems that many files in CUB aren't formatted properly.

Reply:
Note that you can select code sections and then apply formatting to only that selection (Right-Click -> Format Selection)

Reply:
Thanks for the tip!
Are there plans to format the whole codebase and run format checks in CI?

Reply:
Yes there are plans, but it would essentially need a short stop of all work due to massive merge conflicts, so no one has done it yet